dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -164,4 +172,81 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
 
         co_return channelResult;
     }
+
+    Microsoft::Windows::PushNotifications::PushNotificationRegistrationToken PushNotificationManager::RegisterActivator(Microsoft::Windows::PushNotifications::PushNotificationActivationInfo const& details)
+    {
+        winrt::guid taskClsid = details.TaskClsid();
+        DWORD cookie = 0;
+        BackgroundTaskRegistration registeredTask = nullptr;
+
+        if (WI_IsFlagSet(details.Kind(), PushNotificationRegistrationKind::PushTrigger))
+        {
+            bool taskRegistered = false;
+            for (auto task : BackgroundTaskRegistration::AllTasks())
+            {
+                if (task.Value().Name() == backgroundTaskName)
+                {
+                    taskRegistered = true;
+                    break;
+                }
+            }
+
+            if (!taskRegistered)
+            {
+                BackgroundTaskBuilder builder;
+                builder.Name(backgroundTaskName);
+
+                PushNotificationTrigger trigger{};
+                builder.SetTrigger(trigger);
+
+                // QUERY IF API EXISTS (Vb and beyond)
+                if (taskClsid != winrt::guid())
+                {
+                    // Only applicable for a Win32 app
+                    builder.SetTaskEntryPointClsid(taskClsid);
```

Comment:
This API was introduced in Vb. We need an API check. Otherwise we are in trouble.

Reply:
Right now there is a comment as a reminder.

Reply:
Thinking about this a little more...We should fallback to the unpackaged app behavior if this API isn't present. But unpackaged support is not slated for V0.8. So maybe throw NOT_SUPPORTED or something for the preview while we get this in order.

Reply:
Can we add an issupported check for this feature? 

The check would return false if this API is not present or the app is unpackaged for now.

Reply:
https://osgwiki.com/wiki/WinRT_API_Design_Patterns#Type.IsSupported_method